velox/type/StringParser.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <fast_float/fast_float.h>
+#include <folly/Likely.h>
+#include <cmath>
+
+#include "velox/common/base/Status.h"
+
+namespace facebook::velox::util {
+static inline bool characterIsSpace(char c) {
+  return c == ' ' || c == '\t' || c == '\n' || c == '\v' || c == '\f' ||
+      c == '\r';
+}
+
+class StringParser {
```

Comment:
This name is rather generic. Perhaps, rename to StringToFloatParser, then rename stringToFloat method to parse.

Call sites would look like StringToFloatParser::parse